mifosng-provider/src/main/java/org/mifosplatform/portfolio/search/service/SearchReadPlatformServiceImpl.java

```diff
@@ -104,22 +105,22 @@ public String searchSchema(final SearchConditions searchConditions) {
                     + "(s.external_id like :partialSearch and s.external_id not like :search)) ";
 
             final String clientIdentifierExactMatchSql = " (select 'CLIENTIDENTIFIER' as entityType, ci.id as entityId, ci.document_key as entityName, "
-                    + " null as entityExternalId, null as entityAccountNo, c.id as parentId, c.display_name as parentName, c.status_enum as entityStatusEnum "
+                    + " null as entityExternalId, null as entityAccountNo, c.id as parentId, c.display_name as parentName,null as entityMobileNo, c.status_enum as entityStatusEnum "
                     + " from m_client_identifier ci join m_client c on ci.client_id=c.id join m_office o on o.id = c.office_id "
-                    + " where o.hierarchy like :hierarchy and ci.document_key like :search) ";
+                    + " where o.hierarchy like :hierarchy and ci.document_key like :search ) ";
 
             final String clientIdentifierMatchSql = " (select 'CLIENTIDENTIFIER' as entityType, ci.id as entityId, ci.document_key as entityName, "
-                    + " null as entityExternalId, null as entityAccountNo, c.id as parentId, c.display_name as parentName, c.status_enum as entityStatusEnum "
+                    + " null as entityExternalId, null as entityAccountNo, c.id as parentId, c.display_name as parentName,null as entityMobileNo ,c.status_enum as entityStatusEnum "
                     + " from m_client_identifier ci join m_client c on ci.client_id=c.id join m_office o on o.id = c.office_id "
-                    + " where o.hierarchy like :hierarchy and ci.document_key like :partialSearch and ci.document_key not like :search) ";
+                    + " where o.hierarchy like :hierarchy and (ci.document_key like :partialSearch and ci.document_key not like :search))";
 
             final String groupExactMatchSql = " (select IF(g.level_id=1,'CENTER','GROUP') as entityType, g.id as entityId, g.display_name as entityName, g.external_id as entityExternalId, NULL as entityAccountNo "
```

Comment:
ya i am doing this please give some time

On Wed, May 6, 2015 at 3:02 PM, Vishwas Babu A J <notifications@github.com>
wrote:

> In
> mifosng-provider/src/main/java/org/mifosplatform/portfolio/search/service/SearchReadPlatformServiceImpl.java
> <https://github.com/openMF/mifosx/pull/1367#discussion_r29744942>:
>
> >                      + " from m_client_identifier ci join m_client c on ci.client_id=c.id join m_office o on o.id = c.office_id "
> > -                    + " where o.hierarchy like :hierarchy and ci.document_key like :partialSearch and ci.document_key not like :search) ";
> > +                    + " where o.hierarchy like :hierarchy and (ci.document_key like :partialSearch and ci.document_key not like :search))";
> >
> >              final String groupExactMatchSql = " (select IF(g.level_id=1,'CENTER','GROUP') as entityType, g.id as entityId, g.display_name as entityName, g.external_id as entityExternalId, NULL as entityAccountNo "
>
> @keshav10 <https://github.com/keshav10> : A heads up that I had made this
> fix while merging the code.
>
> Could you please provide the following in a separate PR
>
> a) Update API docs to reflect that client Mobile number can be one of the
> search parameters
> b) Add a test case which tests the search API (with different search
> parameters like only client, global search etc), so that the next time, an
> error similar to this one would be caught by the test case (Reach out to
> sachin if you have any questions on writing the test case)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openMF/mifosx/pull/1367/files#r29744942>.
>